## Import libraries

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats
from statsmodels.stats import proportion as proptests

import matplotlib.pyplot as plt
% matplotlib inline

UsageError: Line magic function `%` not found.


## Load data

In [2]:
train_df = pd.read_csv('data/training.csv')
print('Loaded training data: size {}'.format(train_df.shape))
test_df = pd.read_csv('data/test.csv')
print('Loaded testing data: size {}'.format(test_df.shape))
print('----------------')
print('train_df:')
train_df.head()

Loaded training data: size (84534, 10)
Loaded testing data: size (41650, 10)
----------------
train_df:


ID Promotion  purchase  V1         V2        V3  V4  V5  V6  V7
0   1        No         0   2  30.443518 -1.165083   1   1   3   2
1   3        No         0   3  32.159350 -0.645617   2   3   2   2
2   4        No         0   2  30.431659  0.133583   1   1   4   2
3   5        No         0   0  26.588914 -0.212728   2   1   4   2
4   8       Yes         0   3  28.044331 -0.385883   1   1   2   2

## Calculate IRR and NIR

In [3]:
purchaser_treatment = train_df.groupby('Promotion')['purchase'].sum()[1]
customer_treatment = train_df['Promotion'].value_counts()[0]
purchaser_control = train_df.groupby('Promotion')['purchase'].sum()[0]
customer_control = train_df['Promotion'].value_counts()[1]

IRR = (purchaser_treatment/customer_treatment) - (purchaser_control/customer_control)
print('IRR = {}'.format(IRR))

NIR = (10*purchaser_treatment - 0.15*customer_treatment) - 10*purchaser_control
print('NIR = {}'.format(NIR))

IRR = 0.009454547819772702
NIR = -2334.5999999999995


## Hypothesis

### Hypothesis 1

* H0: $numberCustomerTreatment = numberCustomerControl$
* Ha: $numberCustomerTreatment \ne numberCustomerControl$

$\alpha$ = 0.05 (significance level)


### Test hypothesis 1

Whether the difference between number of customers in treatment group and control group is statistically significant.

In [7]:
totalCustomer = train_df.shape[0]
numberCustomerTreatment = customer_treatment
numberCustomerControl = customer_control

print('Total customers: {}'.format(totalCustomer))
print('Number of customers in treatment group: {}'.format(numberCustomerTreatment))
print('Number of customers in control group: {}'.format(numberCustomerControl))

Total customers: 84534
Number of customers in treatment group: 42364
Number of customers in control group: 42170


In [8]:
# Compute p-value
p = 0.5
sd = np.sqrt(p * (1-p) * totalCustomer)

z_score = ((numberCustomerControl + 0.5) - p * totalCustomer) / sd
p_value = 2*stats.norm.cdf(z_score)

print("z-score: {z}".format(z=z_score))
print("p-value from z-score: {p}".format(p=p_value))

z-score: -0.6638066506192843
p-value from z-score: 0.5068140685419046


Because `p_value = 0.5068` > $\alpha$ = 0.05, the null hypothesis is accepted. That means there's no statistical difference in the sampling population. If it's statistical significant, this will require us to revise random assignment procedures and re-do data collection. This hypothesis testing is to make sure our inferences on the desired metrics are not founded on bias due to sampling population.

### Hypothesis 2

* H0: $numberParticipantTreatment = numberParticipantControl$
* Ha: $numberParticipantTreatment \ne numberParticipantControl$



### Test hypothesis 2

Test statistical significant to see whether there's a difference between treatment group and control group